packages/vue-component-library/src/lib-components/FilterSelections.vue

```diff
@@ -0,0 +1,273 @@
+        }
+        else {
+          // Toggle is not in filtered mode, just reset the filtered state but don't close
+          filteredStates.value[filterName] = false
+        }
+      }
+    }
+    else {
+      // Animate height if needed
+      const filterIndex = getFilterIndex(filterName)
+      if (filterIndex >= 0 && toggleRefs.value[filterIndex]?.isOpened && filteredStates.value[filterName])
+        await animateHeightChange(filterIndex)
+    }
+  }
+  else {
+    // Add option
+    currentSelection.push(option.value)
+    emit('option-selected', filterName, option)
+  }
+
+  // Update router query parameters based on current selections
+  const nextQuery: Record<string, any> = { ...route.query }
+
+  // For each filter, map selections to its facetField in query
+  props.filters.forEach((filter) => {
+    const key = getFilterKey(filter)
+    const facet = filter.facetField
+    if (!facet)
+      return
+
+    const selections = selectedOptions.value[key] || []
+    if (selections.length > 0)
+      nextQuery[facet] = selections
+    else
+      delete nextQuery[facet]
+  })
+
+  // Reset pagination param if present
+  if ('page' in nextQuery)
+    delete nextQuery.page
+
+  router.push({ query: nextQuery })
+}
+</script>
+
+<template>
+  <div :class="classes">
+    <div class="filter-list">
+      <EffectSlideToggle
+        v-for="filter in props.filters"
+        :key="filter.name"
+        ref="toggleRefs"
+        class="filter-item"
+        :prevent-close="hasSelectedOptions(getFilterKey(filter))"
+      >
+        <!-- SUMMARY -->
+        <template #summary>
+          <div
+            class="filter-summary"
+            :class="{ 'is-filtered': filteredStates[getFilterKey(filter)] }"
+            @click="() => onSummaryClick(getFilterKey(filter))"
+          >
+            <span class="filter-name">{{ filter.name || filter.slotName }}</span>
+            <span class="filter-chevron">
+              <SvgFilterIcon aria-hidden="true" />
+            </span>
+          </div>
+        </template>
+
+        <!-- CONTENT -->
+        <div class="filter-content">
+          <slot
+            :name="getFilterKey(filter)"
+            :filter="filter"
+            :selected-options="selectedOptions[getFilterKey(filter)] || []"
+            :toggle-option="createToggleOptionHandler(getFilterKey(filter))"
+          >
+            <!-- Default content if no slot is provided -->
+            <div
+              v-if="filter.options && filter.options.length > 0"
+              class="filter-options"
+            >
+              <TransitionGroup
+                name="filter-option"
+                tag="div"
+                class="filter-options-list"
+              >
+                <label
+                  v-for="option in getFilterOptions(filter)"
+                  :key="option.value"
+                  class="filter-option"
+                  :class="{ 'is-selected': isOptionSelected(getFilterKey(filter), option.value) }"
```

Comment:
You can move that logic into a function:

function optionClasses(filterKey: string, optionValue: string) {
  return {
    
   'is-selected': isOptionSelected(filterKey, optionValue),
  }
}


Then in your template:
```
<label :class="optionClasses(getFilterKey(filter), option.value)">

```